### PR TITLE
MRG, DOC: Mailing list, Gitter -> Discourse forum

### PR DESCRIPTION
**Pull request by hoechenberger**
Replace all mentions of the mailing list and Gitter with references to the new Discourse forum.

Closes #8429

### CODE REVIEW

**Comment by hoechenberger** [comment]
Good to go from my end.

**Review by larsoner — APPROVED** [review]
LGTM, should we wait to merge until we've migrated all emails from the listserv?

File: README.rst [hunk 1]
```diff
@@ -115,10 +115,10 @@ Please see the documentation on the MNE-Python homepage:
 https://mne.tools/dev/install/contributing.html
 
 
-Mailing list
+Forum
 ^^^^^^^^^^^^
```
Comment (agramfort) [inline README.rst:119]:
```suggestion
^^^^^^
```

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
> Can we even migrate the remaining messages without losing our current forum settings?

Sorry I actually meant the *email addresses* not the correspondences themselves.

But as for the content I think we can 1) backup / export the `mne.discourse.group` data / settings / all messages completely, 2) import it on my or someone else's local instance, 3) do whatever [admin gymnastics](https://meta.discourse.org/t/importing-mailing-lists-mbox-listserv-google-groups-emails/79773) are required there to add messages that we missed between the initial import and "final switch time" (ultimately what gets imported are just text files so it shouldn't be too terrible), 3) export the local data and 4) ask the Discourse people to overwrite our `mne.discourse.group` with the exported data.

**Comment by hoechenberger** [comment]
> Sorry I actually meant the _email addresses_ not the correspondences themselves.

Is that really **that** important? Even if users claim their "staged" email addresses, they'll essentially have to register and set a password, if I'm not mistaken. I don't really see much of an advantage here compared to a regular new registration

**Comment by larsoner** [comment]
I assumed we could somehow mass-register everyone and have them just start getting Discourse emails rather than MGH ones. But if they have to register anyway I agree there isn't any point.

**Comment by agramfort** [comment]
get all email subscribers would be great to keep the user stats on list
(for grants...)

>

**Comment by hoechenberger** [comment]
> I assumed we could somehow mass-register everyone and have them just start getting Discourse emails rather than MGH ones. But if they have to register anyway I agree there isn't any point.

linking emails / users to their list postings would be nice indeed

**Review by drammock — APPROVED** [review]

**Comment by cbrnr** [comment]
So does this mean we will wait until 0.23 before we switch? I'd be in favor of doing it sooner than later, so maybe we should backport this?

**Comment by hoechenberger** [comment]
+10 for backport

**Pull request merged**